### ISSUE TITLE
Missing example on container usage

### ISSUE THREAD

**Issue by marema31** [richterger/Perl-LanguageServer#70]
I've managed to make the extension running within a running container by tweaking the `bin/shell-into-appserver.sh` example described in the documentation, the linting and error checking works, but I'm not able to determine  a correct launch.json file with the mapping magic. Can you add a such example in the documentation please.

**Comment by richterger** [comment]
Sorry, the container docs are contributed by an user. I personaly don't use it within containers. 
If you describe your problem more in detail, I might be able to help you.

**Comment by geraldspreer** [comment]
> I've managed to make the extension running within a running container by tweaking the `bin/shell-into-appserver.sh` example described in the documentation, the linting and error checking works, but I'm not able to determine a correct launch.json file with the mapping magic. Can you add a such example in the documentation please.

Would you be willing to post the modified `bin/shell-into-appserver.sh` here? I am trying to get the language-server to work in Docker as well. Maybe we can figure it out together?

**Comment by marema31** [comment]
Since I do not use docker compose, I've just modified it to use directly the docker exec command:
``
#!/usr/bin/env bash
COMMAND=$(echo "$@" | sed 's/^.*perl /perl /')
docker exec -i  the_container $COMMAND

**Comment by marema31** [comment]
One things I do not understand well, the $COMMAND contains `perl -MPerl::LanguageServer -e Perl::LanguageServer::run -- --port 13603 --log-level 2 --version 2.1.0` does it means it's opening a listening port used by the vscode extension, in this case since this port is not published at docker run it is normal that is not working.

**Comment by geraldspreer** [comment]
@marema31 Okay I see. I have got `bin/shell-into-appserver.sh` working with docker-compose. However I have trouble installing Perl::LanguageServer inside the container. But this has nothing to do with the LanguageServer, it's because of my particular project. 
I have not used Docker much but I guess you are right with the open ports. Maybe I start a new Perl project inside an Docker container and try it with just some modules installed.

**Comment by geraldspreer** [comment]
I have managed to get the Language Server running inside the Docker container and VSCode is connecting correctly. However I get this error:
```
Bareword "JSON::true" not allowed while "strict subs" in use at /usr/local/lib/perl5/site_perl/5.24.1/Perl/LanguageServer.pm line 260.
Bareword "JSON::false" not allowed while "strict subs" in use at /usr/local/lib/perl5/site_perl/5.24.1/Perl/LanguageServer.pm line 273.
Type of arg 1 to Coro::async must be block or sub {} (not reference constructor) at /usr/local/lib/perl5/site_perl/5.24.1/Perl/LanguageServer.pm line 305, near "} ;"
BEGIN not safe after errors--compilation aborted at /usr/local/lib/perl5/site_perl/5.24.1/Perl/LanguageServer.pm line 317.
Compilation failed in require.
BEGIN failed--compilation aborted.
```
After that the LanguageServer tries to restart four more times and is stopped. 
@richterger Do you have any suggestions?

**Comment by richterger** [comment]
These errors looks strange. Are the modules JSON and Coro installed inside the container?

**Comment by geraldspreer** [comment]
Yes, Coro and JSON are installed. I have installed Perl::LanguageServer with `cpanm --installdeps`

**Comment by richterger** [comment]
Do you see any errors when you run `perl -MPerl::LanguageServer -e Perl::LanguageServer::run -- --port 13603 --log-level 2`
P.S. The port is only necessary for the debugger

**Comment by geraldspreer** [comment]
@richterger This is the error I get:
```
perl -MPerl::LanguageServer -e Perl::LanguageServer::run -- --port 13603 --log-level 2
Bareword "JSON::true" not allowed while "strict subs" in use at /usr/local/lib/perl5/site_perl/5.24.1/Perl/LanguageServer.pm line 260.
Bareword "JSON::false" not allowed while "strict subs" in use at /usr/local/lib/perl5/site_perl/5.24.1/Perl/LanguageServer.pm line 273.
Type of arg 1 to Coro::async must be block or sub {} (not reference constructor) at /usr/local/lib/perl5/site_perl/5.24.1/Perl/LanguageServer.pm line 305, near "} ;"
BEGIN not safe after errors--compilation aborted at /usr/local/lib/perl5/site_perl/5.24.1/Perl/LanguageServer.pm line 317.
Compilation failed in require.
BEGIN failed--compilation aborted.

```

**Comment by richterger** [comment]
Please try to install either JSON::PP or JSON::XS manually

**Comment by geraldspreer** [comment]
Thank you @richterger for looking into this. I have found the solution, and there nothing wrong on your end. The problem is that the app that I have inherited includes some modules, including JSON::PP, hard into a local folder and sets PERL5LIB to load this local module. JSON::PP here is 0.9x. 
But when I adjust the paths in PERL5LIB to use the latest JSON::PP the language-server installs and runs without any problems. So it's all good. Again, thanks for looking into this.
